pkg/resource/replication_group/custom_update_api.go

```diff
@@ -101,6 +101,28 @@ func (rm *resourceManager) CustomModifyReplicationGroup(
 		return rm.decreaseReplicaCount(ctx, desired, latest)
 	}
 
+	// If there is a scale up modification, then we would prioritize it
+	// over increase/decrease shards. This is important since performing
+	// scale in without scale up might fail due to insufficient memory.
+	if delta.DifferentAt("Spec.CacheNodeType") && desired.ko.Spec.ReplicationGroupID != nil {
+		input := &svcsdk.ListAllowedNodeTypeModificationsInput{}
+		input.SetReplicationGroupId(*desired.ko.Spec.ReplicationGroupID)
+		resp, apiErr := rm.sdkapi.ListAllowedNodeTypeModifications(input)
+		rm.metrics.RecordAPICall("READ_MANY", "ListAllowedNodeTypeModifications", apiErr)
+
+		if apiErr != nil {
+			return nil, apiErr
+		}
```

Comment:
The API call can be avoided here since https://github.com/aws-controllers-k8s/elasticache-controller/blob/main/pkg/resource/replication_group/custom_set_output.go#L130-L138 populates the `ko.Status.AllowedScaleDownModifications` and `ko.Status.AllowedScaleUpModifications` into `latest` object.

Reply:
Thank you! this simplifies the implementation and avoids additional API call.